src/streamdbConsumerClient/httpClient.ts

```diff
@@ -0,0 +1,77 @@
+import {Offset, StreamEvent} from "../model";
+import {Axios} from "axios";
+import {
+  GetStateTransitionsResponse,
+  StreamStateTransitionsResponse
+} from "../gen/stream_consumer/v1alpha1/stream_consumer";
+import {stateTransitionProtoToStreamEvent} from "../streamdb/converters";
+import {PausableStream, StreamController} from "../lib/pausableStream";
+import {ExponentialBackoff, WebsocketBuilder} from "websocket-ts";
+import {StreamDBConsumerClient} from "./consumerClient";
+
+export class StreamDBConsumerHttpClient implements StreamDBConsumerClient {
+  private readonly client: Axios;
+
+  constructor(private readonly uri: string) {
+    this.client = new Axios({
+      baseURL: this.uri,
+      validateStatus: status =>
+        (status >= 200 && status < 300) || status == 404,
+    });
+  }
+
+  public getEvents(
+    stream: string,
+    offset: Offset,
+    count: number,
+    direction: "next" | "last"
+  ): Promise<StreamEvent[]> {
+    return this.client
+      .post(`/api/consumer/${stream}/transitions`,
+        JSON.stringify({
+            offset: offset,
+            count: count,
+            direction: direction.toUpperCase(),
+          }, (key, value) => (typeof value === "bigint" ? value.toString() : value)
+        )
+      ).then(resp =>
+        (JSON.parse(resp.data) as GetStateTransitionsResponse)
+          .stateTransitions
+          .map(stateTransitionProtoToStreamEvent)
+      );
+  }
+
+  public getEventsStream(
+    stream: string,
+    offset: Offset,
+    controller?: StreamController
+  ): PausableStream<StreamEvent> {
+    return PausableStream.create<StreamEvent>((observer, _) => {
```

Comment:
@ivanbenko is it necessary to use PauseState here? I haven't dove deeply into its internals

Reply:
yes, `pauseState` is important. Client should react to its state. It's being used by Buffered Reader to prevent overconsuming events